### PR TITLE
Jetty 12 with EE8

### PR DESCRIPTION
**Pull request by ddworak**

### CODE REVIEW

File: rest/.jvm/src/test/scala/io/udash/rest/EndpointsIntegrationTest.scala [hunk 1]
```diff
@@ -23,16 +24,21 @@ class EndpointsIntegrationTest extends UdashSharedTest with BeforeAndAfterAll wi
...
+  contextHandler.setSessionHandler(new SessionHandler)
+  contextHandler.addServlet(holder, s"$contextPrefix/*")
+
+  val context = new GzipHandler()
+  context.setHandler(contextHandler)
+
+  val server = new Server(port)
   server.setHandler(context)
+  // Unsafe URI compliance is required for testing purposes
+  server.getConnectors.head.getConnectionFactory(classOf[HttpConnectionFactory]).getHttpConfiguration.setUriCompliance(UriCompliance.UNSAFE)
```
Comment (freddie713) [inline rest/.jvm/src/test/scala/io/udash/rest/EndpointsIntegrationTest.scala:41]: URICompliance in jetty 10 looked more or less like this: 
`new UriCompliance("LEGACY_DEFAULT", JEnumSet(Violation.AMBIGUOUS_PATH_SEPARATOR, Violation.AMBIGUOUS_PATH_ENCODING))`
Are you sure we want to allow more violations?

File: guide/backend/src/main/scala/io/udash/web/server/ApplicationServer.scala [hunk 2]
```diff
@@ -49,53 +55,30 @@ class ApplicationServer(val port: Int, homepageResourceBase: String, guideResour
...
-    val context = new ServletContextHandler(ServletContextHandler.SESSIONS)
-    context.insertHandler(new GzipHandler)
-    context.setVirtualHosts(hosts)
-    context
+  server.setHandler(
+    new RewriteHandler(new ContextHandlerCollection().setup(_.setHandlers(homepage, guide)))
+      .setup(_.addRule(new RewriteRegexRule("^/(?!assets|scripts|styles|atm|rest_api)(.*/?)*$", "/")))
+  )
+
+  private def createContextHandler(hosts: Array[String], resourceBase: String): ServletContextHandler = {
```
Comment (freddie713) [inline guide/backend/src/main/scala/io/udash/web/server/ApplicationServer.scala:75]: I would consider adding normalization to resourceBase as jetty 12 changed implementation of SymlinkAllowedResourceAliasChecker and it no longer allows resources with /../ in path resulting in 404 errors.

File: rest/.jvm/src/test/scala/io/udash/rest/EndpointsIntegrationTest.scala (hunk 1 above)
Comment (ddworak) [reply rest/.jvm/src/test/scala/io/udash/rest/EndpointsIntegrationTest.scala:41]: It was just for tests but I'll use the legacy one, thanks.

**Review by freddie713 — APPROVED** [review]
LGTM

**Pull request merged**